### PR TITLE
Only count VDIs on tested SRs

### PR DESCRIPTION
**Pull request by benjamreis**
When using `-sr` or `-default-sr` only
count VDIs on those SR to check for leak
VDIs.

### CODE REVIEW

**Comment by psafont** [comment]
Please use structural equality instead of the one that checks for the same memory location :)

**Comment by benjamreis** [comment]
I've been manually tested this PR and here's the behaviours:
- Testing on SRA + import VM on SRB => no leak VDIS
- Testing on SRA + import VM on SRA => leak VDIS detected
- Testing on default SR + import VM on another SR => no leak VDIS
- Testing on default SR + import VM on default SR => leak VDIS detected
- Testing on all SRs + import a VM on any SR => leak VDIs detected

Seems correct from my POV

File: ocaml/quicktest/qt_filter.ml [hunk 1]
```diff
@@ -25,31 +25,49 @@ let count_vdis rpc session_id sr =
...
 let init () =
   session_id := Qt.init_session !A.rpc !A.username !A.password ;
+  let test_sr_uuid = get_test_sr_uuid () in
   Client.Client.SR.get_all_records ~rpc:!A.rpc ~session_id:!session_id
   |> List.iter (fun (ref, sr) ->
-         if List.mem `scan sr.API.sR_allowed_operations then
-           let before = count_vdis !A.rpc !session_id ref in
-           Hashtbl.add vdi_count sr.API.sR_uuid before
+         if test_sr_uuid = "" || sr.API.sR_uuid = test_sr_uuid then
+           if List.mem `scan sr.API.sR_allowed_operations then
```
Comment (lindig) [inline ocaml/quicktest/qt_filter.ml:50]: The two `if` conditions could be folded into one.

File: ocaml/quicktest/qt_filter.ml [hunk 2]
```diff
@@ -25,31 +25,49 @@ let count_vdis rpc session_id sr =
...
      )
 
 (** Called at the end of the quicktests to check that no resources leaked
     during the test run *)
 let finish () =
+  let test_sr_uuid = get_test_sr_uuid () in
   Client.Client.SR.get_all_records ~rpc:!A.rpc ~session_id:!session_id
   |> List.iter (fun (ref, sr) ->
          match Hashtbl.find_opt vdi_count sr.API.sR_uuid with
          | Some before ->
```
Comment (lindig) [inline ocaml/quicktest/qt_filter.ml:62]:
```
| Some before when test_sr_uuid = "" || sr.API.sR_uuid = test_sr_uuid -> ...
| Some _ -> (* is this a special case that needs to be logged? *)
```

**Review by lindig — APPROVED** [review]

**Review by psafont — APPROVED** [review]
Quicktest passes in Xenserver's internal tests using this PR

**Pull request merged**